My teams tab.

symbolist opened the pull request:
https://openedx.atlassian.net/browse/TNL-1911

Show the user's teams on the "My Teams" tab.

Sandbox: http://myteams.sandbox.edx.org

symbolist commented:
@explorerleslie The sandbox is up. Since it is not possible to join teams via the UI, I have added the staff user to a couple of teams through the django shell. The list can be viewed on: http://myteams.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/teams/#teams

@marcotuts I did not see the need for any sass changes for this. But can you give the UI a quick look to see if everything is in order? The teams cards are exactly the same as those on the "Browse" tab.

@cptvitamin Can you also review? Though the "My Teams" tab uses the same templates as on the "Browse" tab, so hopefully everything should work from the accessibility point of view as expected already.

explorerleslie commented:
@symbolist is the "view" button supposed to take you through to the Team's page? it's not doing that right now.

symbolist commented:
@explorerleslie I am working off master, and that is being worked on in https://github.com/edx/edx-platform/pull/9105. I'll rebase when it merges.

marcotuts commented:
:+1:

explorerleslie commented:
:+1:

symbolist commented:
@muzaffaryousaf This is ready for a review from you.

muzaffaryousaf commented on lms/djangoapps/teams/static/teams/js/views/teams_tab.js:
```diff
@@ -60,23 +64,34 @@
+                        this.teamMemberships,
+                        {
+                            url: this.teamMembershipsUrl,
+                            course_id: this.courseID,
```
courseID instead of course_id ?

muzaffaryousaf commented on lms/djangoapps/teams/models.py:
```diff
@@ -88,3 +88,23 @@ class Meta(object):
     date_joined = models.DateTimeField(auto_now_add=True)
+
+    @classmethod
+    def get_memberships(cls, username=None, course_ids=None, team_id=None):
```
Python unit-tests are missing for this.

muzaffaryousaf commented on lms/djangoapps/teams/views.py:
```diff
@@ -91,18 +93,28 @@ def get(self, request, course_id):
         )
         user = request.user
+
+        team_memberships = CourseTeamMembership.get_memberships(request.user.username, [course.id])
```
Python unit-tests are missing for this too.

symbolist replied:
Yes, I'll add a unit test for this.

muzaffaryousaf commented:
@symbolist I"m seeing unexpected behaviour for my teams. Here are the steps to see:
* You are on My Teams page and you see `Showing 0 out of 0 total`.
*  Now you have created a team successfully, this will take you to team detail page. (By default you are member of this team)
* Click on `All Topics` in breadcrumb that will show you browse tab.
* Click on `My Teams` tab and you are still seeing `Showing 0 out of 0 total` without any team card.
* When you refresh the page then you will see your created team.

muzaffaryousaf commented:
@symbolist I'm done with review, things are in good shape except the bug and minor nits which mentioned in my comments. Thanks.

symbolist commented:
@muzaffaryousaf Yes, currently the "My Teams" tab is not refreshed when a user joins or leaves a team. I wasn't able to get it to work without the PRs which add the add/leave functionality. Its a small change, so we can tackle it separately.

symbolist commented:
@muhammad-ammar Can you also give this a review?

muhammad-ammar commented:
@symbolist shouldn't route be `my-teams` instead of `teams`?

muhammad-ammar commented:
@symbolist I am done with the review. lgtm.  Besides route change :+1:

symbolist commented:
@muzaffaryousaf @muhammad-ammar I have addressed your comments. The tab hash is now "my-teams" instead of just "teams". And I have rebased and squashed commits.

muzaffaryousaf commented:
:+1:
